editor/debugger/editor_debugger_inspector.cpp

```diff
@@ -178,40 +254,212 @@ ObjectID EditorDebuggerInspector::add_object(const Array &p_arr) {
 			}
 		}
 
-		//always add the property, since props may have been added or removed
+		// Always add the property, since props may have been added or removed.
 		debug_obj->prop_list.push_back(pinfo);
 
 		if (!debug_obj->prop_values.has(pinfo.name)) {
 			new_props_added++;
 			debug_obj->prop_values[pinfo.name] = var;
-		} else {
-			if (bool(Variant::evaluate(Variant::OP_NOT_EQUAL, debug_obj->prop_values[pinfo.name], var))) {
-				debug_obj->prop_values[pinfo.name] = var;
-				changed.insert(pinfo.name);
-			}
+		} else if (bool(Variant::evaluate(Variant::OP_NOT_EQUAL, debug_obj->prop_values[pinfo.name], var))) {
+			debug_obj->prop_values[pinfo.name] = var;
+			changed.insert(pinfo.name);
 		}
 	}
 
 	if (old_prop_size == debug_obj->prop_list.size() && new_props_added == 0) {
-		//only some may have changed, if so, then update those, if exist
+		// Only some may have changed, if so, then update those, if they exist.
 		for (const String &E : changed) {
 			emit_signal(SNAME("object_property_updated"), debug_obj->remote_object_id, E);
 		}
 	} else {
-		//full update, because props were added or removed
+		// Full update, because props were added or removed.
 		debug_obj->update();
 	}
 	return obj.id;
 }
 
+EditorDebuggerMultiRemoteObject *EditorDebuggerInspector::get_multi_objects(const Array &p_arr) {
```

Comment:
Some code duplication with MultiNodeEdit.

Reply:
I know, not much I can do about that.

Reply:
EditorDebuggerRemoteObject could be merged with EditorDebuggerMultiRemoteObject. The latter should be able to handle single object.

Reply:
That goes back to my initial request for feedback (https://github.com/godotengine/godot/pull/99680#discussion_r1856786606) about merging the "inspect_multi_objects" and "inspect_object" messages.

Reply:
I missed it. I don't think it makes sense to have 2 messages for basically the same.